difflint/lib.py

```diff
@@ -205,6 +229,73 @@ def detect_new_diff_lint_errors(log_output):
     return False
 
 
+def save_merge_state():
+    """
+    Saves the merging state in the event that we're resolving
+    a merge conflict. Stashing will mangle the merge state unless
+    we have saved it ahead of time. The reverse operation to this
+    function is load_merge_state().
+
+    Special thanks to the overcommit project for solving this
+    problem in a similar context. (https://github.com/brigade/overcommit)
+
+    Inputs: None
+
+    Output: If we were are in a merge conflict state, this will return a
+            tuple containing first the merge message, and second the
+            merge commit hash.
+            If we were not in a merge conflict state, this will return a
+            tuple containing first an empty string, and second another
+            empty string.
+    """
+    merge_head_commit_hash = ''
+    try:
+        with open(devnull, 'w') as DEV_NULL:
+            # This will fail if MERGE_HEAD doesn't exist.
+            merge_head_commit_hash = check_output(['git', 'rev-parse', 'MERGE_HEAD'],
+                                                  stderr=DEV_NULL).decode()
+    except CalledProcessError:
+        return ('', '')
+
+    merge_msg = ''
+    merge_msg_filename = make_path_from_cwd('.git', 'MERGE_MSG')
```

Comment:
No need to call `make_path_from_cwd` here. You can just pass '.git/MERGE_MSG' directly to `os.path.isfile`.